src/components/Header.tsx

```diff
@@ -0,0 +1,71 @@
+import { Component } from 'react';
+import SearchBar from './SearchBar.tsx';
+import { getPokemonList, type Pokemon } from '../services/api.service.tsx';
+import '../styles/Header.scss';
+import LogoImage from '../assets/logo.png';
+import {
+  LocalStorageService,
+  STORAGE_KEYS,
+} from '../services/local-storage.service.tsx';
+
+interface HeaderProps {
+  setPokemons: (data: Pokemon[]) => void;
+  setIsLoading: (data: boolean) => void;
+}
+
+interface HeaderState {
+  searchValue: string;
+}
+
+class Header extends Component<HeaderProps, HeaderState> {
+  constructor(props: HeaderProps) {
+    super(props);
+    const savedSearch =
+      LocalStorageService.getSearchTerm(STORAGE_KEYS.searchTerm) || '';
+    this.state = {
+      searchValue: savedSearch,
+    };
+    if (savedSearch) {
+      this.handleSearch(savedSearch);
+    }
+  }
```

Comment:
a constructor is not needed here, but `componentDidMount` is needed
```suggestion
  state: HeaderState = {
    searchValue:
      LocalStorageService.getSearchTerm(STORAGE_KEYS.searchTerm) || '',
  };
  
   componentDidMount() {
    this.handleSearch(this.state.searchValue);
  }
```